Content.Server/AI/Operators/Inventory/OpenStorageOperator.cs

```diff
@@ -0,0 +1,55 @@
+using Content.Server.AI.Utility;
+using Content.Server.AI.WorldState.States.Inventory;
+using Content.Server.GameObjects.Components;
+using Content.Server.GameObjects.EntitySystems;
+using Robust.Shared.Containers;
+using Robust.Shared.Interfaces.GameObjects;
+
+namespace Content.Server.AI.Operators.Inventory
+{
+    /// <summary>
+    /// If the target is in EntityStorage will open its parent container
+    /// </summary>
+    public sealed class OpenStorageOperator : AiOperator
+    {
+        private readonly IEntity _owner;
+        private readonly IEntity _target;
+        
+        public OpenStorageOperator(IEntity owner, IEntity target)
+        {
+            _owner = owner;
+            _target = target;
+        }
+        
+        public override Outcome Execute(float frameTime)
+        {
+            if ((_target.Transform.GridPosition.Position - _owner.Transform.GridPosition.Position).Length >
+                SharedInteractionSystem.InteractionRange)
+            {
+                return Outcome.Failed;
+            }
+
+            if (!ContainerHelpers.TryGetContainer(_target, out var container))
+            {
+                return Outcome.Success;
```

Comment:
Why is this a success?

Reply:
Couldn't think of an easier way to handle grabbing items from lockers / crates without having duplicate pickup code. As it is now it's part of "GoPickupEntitySequence" so any actions that require picking an item up don't need to worry about handling entitystorage.